components/eamxx/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -71,7 +71,9 @@ void AtmosphereProcess::initialize (const TimeStamp& t0, const RunType run_type)
     start_timer (m_timer_prefix + this->name() + "::init");
   }
 
-  log (LogLevel::info,"  Initializing " + name() + "...");
+  if (this->type()!=AtmosphereProcessType::Diagnostic) {
+    log (LogLevel::info,"  Initializing " + name() + "...");
+  }
   m_atm_logger->flush(); // During init, flush often (to help debug crashes)
```

Comment:
You should prob move the flush call inside the if statement, to avoid a pointless file sync. Even if nothing was logged since the last flush, there may be a nontrivial cost in case of tens of diags and slow filesystems (I am not entirely sure, since I don't know how spdlog handles flush calls, but we may as well avoid the flush).